common/scripts/changelog/collect.mjs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * Script to collect CHANGELOG files for stable and beta releases.
+ *
+ * This script processes the change files generated previously via the ./change.mjs.
+ * It collects the changes into the CHANGELOG relevant for the release type.
+ *
+ * Usage:
+ *   node common/scripts/changelog/collect.mjs stable
+ *   node common/scripts/changelog/collect.mjs beta
+ */
+
+import { copyFile, rm, rename } from 'fs/promises';
+import { exec } from "../lib/exec.mjs";
+import { CHANGE_DIR, CHANGE_DIR_BETA, CHANGE_DIR_STABLE_TEMP, COMMUNICATION_REACT_CHANGELOG_BETA, COMMUNICATION_REACT_CHANGELOG_STABLE, COMMUNICATION_REACT_CHANGELOG_TEMPORARY } from './constants.mjs';
+import { generateChangelogs } from './changelog.mjs';
+
+async function main() {
+    const args = process.argv;
+    const buildFlavor = args[2];
+
+    await ensureCleanWorkingDirectory();
+    try {
+        if (buildFlavor === 'beta') {
+            await collectionBetaChangelog();
+        } else if (buildFlavor === 'stable') {
+            await collectionStableChangelog();
+        } else {
+            console.error(`Usage: ${args[0]} ${args[1]} <beta|stable>`);
+            throw new Error(`Unknown build flavor ${buildFlavor}`);
+        }
+    } finally {
+        await restoreWorkingDirectory()
+    }
+}
+
+async function collectionBetaChangelog() {
+    await swapInBetaChangeFiles();
+    await createTemporaryChangelog(COMMUNICATION_REACT_CHANGELOG_BETA);
+    await generateChangelogs();
+    await restoreStableChangeFiles();
+    await commitChangelog(COMMUNICATION_REACT_CHANGELOG_BETA);
+}
+
+async function collectionStableChangelog() {
+    await createTemporaryChangelog(COMMUNICATION_REACT_CHANGELOG_STABLE);
+    await generateChangelogs();
+    await commitChangelog(COMMUNICATION_REACT_CHANGELOG_STABLE);
+}
+
+async function swapInBetaChangeFiles() {
+    await rm(CHANGE_DIR_STABLE_TEMP, { recursive: true, force: true })
+    await rename(CHANGE_DIR, CHANGE_DIR_STABLE_TEMP);
+    await rename(CHANGE_DIR_BETA, CHANGE_DIR);
+}
+
+async function restoreStableChangeFiles() {
+    await rename(CHANGE_DIR, CHANGE_DIR_BETA);
+    await rename(CHANGE_DIR_STABLE_TEMP, CHANGE_DIR);
+}
+
+async function ensureCleanWorkingDirectory() {
+    try {
+        await exec('git diff --name-status --exit-code');
+        await exec('git diff --name-status --exit-code --cached');
+    } catch (e) {
+        throw new Error(`Detected dirty working directory: ${e}`);
+    }
+}
+
+async function createTemporaryChangelog(target) {
+    await rm(COMMUNICATION_REACT_CHANGELOG_TEMPORARY, { force: true })
+    await copyFile(target, COMMUNICATION_REACT_CHANGELOG_TEMPORARY);
```

Comment:
Is a rm need on line 74? 
Reading copyFile documentation, it says "by default, fs.copyFile() will overwrite the file if it already exists at the given destination."

Reply:
Hmm. Perhaps. I'll keep the `rm` still as it's more readable (you delete, then you copy) and the performance cost isn't an issue.